python/cudf/cudf/core/multiindex.py

```diff
@@ -437,6 +437,7 @@ def take(self, indices):
         if isinstance(indices, (Integral, Sequence)):
             indices = np.array(indices)
         elif isinstance(indices, Series):
+            # leave this alone -- fails 'test_string_repr' test
             indices = indices.to_gpu_array()
```

Comment:
If this test is failing it's likely relying on the behavior of `to_gpu_array()` throwing away nulls which could have incorrect behavior. We should likely assert there's no nulls before this call and then use `indices.data.mem`

Reply:
Thanks so much for bearing with me. @trxcllnt and I paired for a while last night and I think I'm finally getting some steam. This was the most contentious issues. I'll get into handling this per your spec. The ghost of the dumb comment in `dataframe.py` on #397 I'm pretty sure is an artifact of that pairing and `dataframe.py` is a simple fix. Verifying that I've got no nulls here feels hack but might be the truth and correct. 

I've got my test/dev env humming. I'll shoot back with something that works for both but I might stew over this one a bit too long after lunch.

Reply:
Since this is feeding into a gather call it would be super invalid to have nulls, so asserting there's no nulls would be good here, otherwise casting the null to any type of integer would give undefined behavior.

Reply:
The issue was rooted in the the problems with `cupy`/`numb` breaking the return of `Series.values()` somewhat silently. Tests are passing. All instances now pull from `mem` instead of `to_gpu_array`. Pushed the fix with PR #1302 fixing that issue and everything appears to be good.